code/ui/manager/src/components/preview/Preview.tsx

```diff
@@ -107,22 +94,18 @@ const Preview = React.memo<PreviewProps>(function Preview(props) {
         <S.PreviewContainer>
           <ToolbarComp
             key="tools"
-            entry={entry}
-            api={api}
             isShown={showToolbar}
-            tabs={visibleTabsInToolbar}
+            tabId={tabId}
+            tabs={tabs}
+            tools={tools}
+            toolsExtra={toolsExtra}
+            api={api}
           />
           <S.FrameWrap key="frame">
-            <Canvas {...{ withLoader, baseUrl }} />
-            {tabs.map(({ render: Render, match, ...t }, i) => {
-              // @ts-expect-error (Converted from ts-ignore)
-              const key = t.id || t.key || i;
-              return (
-                <Fragment key={key}>
-                  <Location>{(lp) => <Render active={match(lp)} />}</Location>
-                </Fragment>
-              );
-            })}
+            {tabContent && <S.IframeWrapper>{tabContent({ active: true })}</S.IframeWrapper>}
+            <S.CanvasWrap style={!!tabId ? { display: 'none' } : {}}>
```

Comment:
given this is now an actual Emotion component it would be cleaner to pass a `tabId` prop instead of using `style`